java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -260,6 +271,38 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
   }
 
   /**
+   * Return a list of UrlGroups sorted by the rank of the top-ranked pair in each group.
+   * @return a sorted list of UrlGroups.
+   */
+  public List<UrlGroup> getUrlGroupsSortedByRank() {
+    List<PwPair> sortedPwPairs = getPwPairsSortedByRank();
+
+    Map<String, UrlGroup> pwGroupMap = new HashMap<>();
+    for (PwPair pwPair : sortedPwPairs) {
+      PwsResult pwsResult = pwPair.getPwsResult();
+      if (pwsResult != null) {
+        String groupId = pwsResult.getGroupId();
+        if (groupId != null && !groupId.equals("")) {
```

Comment:
..clarification: current PWS codebase will always return a result but the feature has not been deployed to stable.  Can do that once we are comfortable (I'm happy to do it asap but I don't have time to test right now, if anyone wants to run DEV through the test gauntlet just let me know).